Module.php

```diff
@@ -0,0 +1,179 @@
+use Omeka\Api\Manager;
+use Omeka\Api\Representation\ItemRepresentation;
+use Omeka\Api\Representation\ResourceClassRepresentation;
+use Omeka\Api\Request;
+use Omeka\Job\Dispatcher;
+use Omeka\Module\AbstractModule;
+use Omeka\Module\Exception\ModuleCannotInstallException;
+use Omeka\Module\Module as DefaultModule;
+use Omeka\Stdlib\Message;
+use Omeka\Api\Exception\ValidationException;
+
+final class Module extends AbstractModule
+{
+    private ?Dispatcher $dispatcher = null;
+    private $api = null;
+    private const MODULE_DEPENDENCIES = [
+        ['name' => 'CustomVocab', 'version' => '1.7.1'],
+        ['name' => 'AdvancedResourceTemplate', 'version' => '3.4.4.17'],
+        ['name' => 'NumericDataTypes', 'version' => '1.11.0'],
+    ];
+
+    private const FOLDERS = [
+        ['path' => 'files/datacatalogs/'],
+        ['path' => 'files/datadumps/'],
+    ];
+
+    public function getConfig()
+    {
+        return include __DIR__ . '/config/module.config.php';
+    }
+
+    public function setServiceLocator(ServiceLocatorInterface $serviceLocator)
+    {
+        $this->serviceLocator = $serviceLocator;
+    }
+
+    public function install(ServiceLocatorInterface $serviceLocator)
+    {
+
+//        $this->checkPrerequisites($serviceLocator);
+//        $this->createFoldersIfTheyDontExist();
+//        $this->installSchemaOrgVocabulary($serviceLocator);
+//        $this->installCustomVocabularies();
+//        $this->installTemplates();
+    }
+
+    protected function checkPrerequisites(ServiceLocatorInterface $serviceLocator): void
+    {
+        /* @var DefaultModule $module */
+        foreach (self::MODULE_DEPENDENCIES as $moduleDependency) {
+            $module = $serviceLocator->get('Omeka\ModuleManager')
+                ->getModule($moduleDependency['name'])
+            ;
+
+            if ($module && version_compare($module->getIni('version') ?? '', $moduleDependency['version'], '<')) {
+                $translator = $serviceLocator->get('MvcTranslator');
+                $message = new Message(
+                    $translator->translate('This module requires the module "%s", version %s or above.'), // @translate
+                    $moduleDependency['name'], $moduleDependency['version']
+                );
+                throw new ModuleCannotInstallException((string) $message);
+            }
+
+            if ($module->getState() !== 'active') {
+                $translator = $serviceLocator->get('MvcTranslator');
+                $message = new Message(
+                    $translator->translate('The "%s" module must be active'), // @translate
+                    $moduleDependency['name']
+                );
+                throw new ModuleCannotInstallException((string) $message);
+            }
+        }
+    }
+
+    protected function createFoldersIfTheyDontExist(): void
+    {
+        foreach (self::FOLDERS as $folderName) {
+            $this->checkFolder($folderName['path']);
+        }
```

Comment:
We have not yet talked about logging, but this might be a good thing to add this action to the logs? The log level should be quite low. But it might be relevant information to see when auditing a certain request against the logs.

Not one for this PR per se. But one to discuss with the PM at one point